op-challenger/fault/monitor_test.go

```diff
@@ -103,10 +105,76 @@ func TestMonitorOnlyCreateSpecifiedGame(t *testing.T) {
 	require.Equal(t, 1, games.created[addr2].progressCount)
 }
 
+func TestMonitorGames(t *testing.T) {
+	t.Run("subscription errors", func(t *testing.T) {
+		addr1 := common.Address{0xaa}
+		addr2 := common.Address{0xbb}
+		monitor, source, games, _, mockSub := setupMonitorTest(t, []common.Address{})
+
+		source.games = []FaultDisputeGame{
+			{
+				Proxy:     addr1,
+				Timestamp: 9999,
+			},
+			{
+				Proxy:     addr2,
+				Timestamp: 9999,
+			},
+		}
+
+		var err error
+		go func() {
+			err = monitor.MonitorGames(context.Background())
+		}()
+		mockSub.errChan <- ethereum.NotFound
+		require.ErrorIs(t, err, ethereum.NotFound)
+		require.Len(t, games.created, 0, "should not create game agents")
+	})
+
+	t.Run("context cancelled", func(t *testing.T) {
+		monitor, _, _, _, _ := setupMonitorTest(t, []common.Address{})
+		ctx, cancel := context.WithCancel(context.Background())
+		cancel()
+		err := monitor.MonitorGames(ctx)
+		require.ErrorIs(t, err, context.Canceled)
+	})
+
+	t.Run("new head spawns games", func(t *testing.T) {
+		addr1 := common.Address{0xaa}
+		addr2 := common.Address{0xbb}
+		monitor, source, games, _, mockSub := setupMonitorTest(t, []common.Address{})
+
+		source.games = []FaultDisputeGame{
+			{
+				Proxy:     addr1,
+				Timestamp: 9999,
+			},
+			{
+				Proxy:     addr2,
+				Timestamp: 9999,
+			},
+		}
+
+		var err error
+		go func() {
+			err = monitor.MonitorGames(context.Background())
+		}()
```

Comment:
Something will need to trigger this function to exit and then wait for it to complete or we'll leak resources.